pkg/client/application_client.go

```diff
@@ -67,14 +75,15 @@ func (o OkctlApplication) HasService() bool {
 // Validate knows if the application is valid or not
 func (o OkctlApplication) Validate() error {
 	return validation.ValidateStruct(&o,
-		validation.Field(&o.Name, validation.Required, is.DNSName),
-		validation.Field(&o.Namespace, validation.Required, is.DNSName),
-		validation.Field(&o.Image, validation.Required, is.DNSName),
+		validation.Field(&o.Name, validation.Required, is.Subdomain),
+		validation.Field(&o.Namespace, validation.Required, is.Subdomain),
+		validation.Field(&o.Image, validation.Required),
 		validation.Field(&o.Version, validation.Required),
-		validation.Field(&o.ImagePullSecret, validation.Required, is.DNSName),
+		validation.Field(&o.ImagePullSecret, is.Subdomain),
 		validation.Field(&o.SubDomain, is.Subdomain),
-		validation.Field(&o.Port, is.Port),
-		validation.Field(&o.Replicas, validation.Min(0)),
+		validation.Field(&o.Port, validation.Min(lowestPossiblePort), validation.Max(highestPossiblePort)),
+		validation.Field(&o.Replicas, validation.Min(lowestPossibleReplicas)),
```

Comment:
I look forward to creating a deployment with 34567898765434567 replicas 😬 (just trying to be funny, don't change it)

Reply:
yeah! left it open in case one of the big cloud providers decides to migrate to okctl